activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -76,7 +76,7 @@ def associated(*associations)
         associations.each do |association|
           reflection = scope_association_reflection(association)
           @scope.joins!(association)
-          if @scope.values.size > 1
+          if reflection.options[:class_name] || (reflection.options[:class_name] && @scope.values[:extending])
```

Comment:
Is there any case where `reflection.options[:class_name]` is `false` but `(reflection.options[:class_name] && @scope.values[:extending])` is `true`?

I don't get why this isn't only `reflection.options[:class_name] && @scope.values[:extending]`

Reply:
Rafael, thanks for the feedback.  I had initially thought so because of an earlier iteration of this but I just ran it with out the `Or` clause and it works! Just pushed up a version with the `Or` removed.